src/gc/gc.cpp

```diff
@@ -11958,6 +11958,7 @@ BOOL gc_heap::a_fit_segment_end_p (int gen_number,
         else
         {
             dprintf (2, ("can't grow segment, doing a full gc"));
+            dd_new_allocation (dynamic_data_of(gen_number)) += limit;
```

Comment:
this is not necessary; the reason why it failed to allocate is because it wanted to do a gen2 GC but couldn't due to the elevation logic which you fixed in generation_to_condemn. are you seeing that's not enough?